File: rr2/code/rr2/renderer/actionpack-3.1.0.rc6/lib/abstract_controller/logger.rb
#---
# Excerpted from "Rails Recipes",
# published by The Pragmatic Bookshelf.
# Copyrights apply to this code. It may not be used to create training material, 
# courses, books, articles, and the like. Contact us if you are in doubt.
# We make no guarantees that this code is fit for any purpose. 
# Visit http://www.pragmaticprogrammer.com/titles/rr2 for more book information.
#---
require "active_support/core_ext/logger"
require "active_support/benchmarkable"

module AbstractController
  module Logger
    extend ActiveSupport::Concern

    included do
      config_accessor :logger
      extend ActiveSupport::Benchmarkable
    end
  end
end

File: mwjsember/code/ch7/ember-note/app/initializers/logger.js
/***
 * Excerpted from "Deliver Audacious Web Apps with Ember 2",
 * published by The Pragmatic Bookshelf.
 * Copyrights apply to this code. It may not be used to create training material, 
 * courses, books, articles, and the like. Contact us if you are in doubt.
 * We make no guarantees that this code is fit for any purpose. 
 * Visit http://www.pragmaticprogrammer.com/titles/mwjsember for more book information.
***/
import Ember from 'ember';

export function initialize(registry, application) {
  var logger = Ember.Object.extend({
    log: function(message) {
      console.log(message);
    }
  });
  application.register('logger:main', logger);
  application.inject('route','logger','logger:main');  
}

export default {
  name: 'logger',
  initialize: initialize
};
